web/packages/build/.eslintrc.js

```diff
@@ -1,180 +0,0 @@
-/*
- * Teleport
- * Copyright (C) 2023  Gravitational, Inc.
- *
- * This program is free software: you can redistribute it and/or modify
- * it under the terms of the GNU Affero General Public License as published by
- * the Free Software Foundation, either version 3 of the License, or
- * (at your option) any later version.
- *
- * This program is distributed in the hope that it will be useful,
- * but WITHOUT ANY WARRANTY; without even the implied warranty of
- * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
- * GNU Affero General Public License for more details.
- *
- * You should have received a copy of the GNU Affero General Public License
- * along with this program.  If not, see <http://www.gnu.org/licenses/>.
- */
-
-module.exports = {
-  parser: '@typescript-eslint/parser',
-  parserOptions: {
-    ecmaVersion: 6,
-    ecmaFeatures: {
-      jsx: true,
-      experimentalObjectRestSpread: true,
-    },
-  },
-  env: {
-    es6: true,
-    browser: true,
-    node: true,
-  },
-  globals: {
-    expect: true,
-    jest: true,
-  },
-  ignorePatterns: ['**/dist/**'],
-  extends: [
-    'eslint:recommended',
-    'plugin:@typescript-eslint/recommended',
-    'plugin:import/errors',
-    'plugin:import/warnings',
-    'plugin:import/typescript',
-  ],
-  plugins: ['react', 'babel', 'import', 'react-hooks'],
-  overrides: [
-    {
-      files: ['**/*.test.{ts,tsx,js,jsx}'],
-      plugins: ['jest'],
-      extends: [
```

Comment:
We miss this in the new config so these rules don't work. Having a plugin in `plugins` doesn't automatically activate it.

Reply:
https://github.com/gravitational/teleport/blob/avatus/eslint/web/packages/build/eslint.config.mjs#L128-L151

is this not that? it seems to work for me. if i for example change this rule to `error`
```
'jest/consistent-test-it': 'error',
```
`pnpm lint` throws a bunch of a new errors

Reply:
Yeah, but it did it because you explicitly enabled this rule. If you remove the rule completely, you will not get any warnings/errors, which means it is “disabled” by default.

Reply:
I see what you mean. What is your recommendation? If these are "off" but don't affect any defaults, I'm unsure what would be required of us. Should we turn on at least one?

Reply:
I think what Grzegorz means is that the new configuration does not use any configs from jest, jest-dom and testing-library/react. So we'd need to add them in the override and also keep the existing rules that are in the override.

Reply:
I'm not following. in the new config schema, they suggest to do it this way
https://www.npmjs.com/package/eslint-plugin-jest

with examples of using override for the _old_ eslint. but in the new one, they say that adding in rules/plugins to the object is just fine.

```
  {
    // update this to match your test files
    files: ['**/*.spec.js', '**/*.test.js'],
    plugins: { jest: pluginJest },
    languageOptions: {
      globals: pluginJest.environments.globals.globals,
    },
    rules: {
      'jest/no-disabled-tests': 'warn',
      'jest/no-focused-tests': 'error',
      'jest/no-identical-title': 'error',
      'jest/prefer-to-have-length': 'warn',
      'jest/valid-expect': 'error',
    },
  },
```

Reply:
Uhhh I have it working. At first the new syntax was confusing to me, but actually it's quite simple. 
If possible, we should use the entire recommended config from a plugin (like `eslint.configs.recommended`) which will merge everything automatically. If we need to apply a few plugins for a single file pattern, we need to this manually, like:
```ts
files: ['**/*.test.{ts,tsx,js,jsx}'],
plugins: {
  jest: jestPlugin,
  'testing-library': testingLibraryPlugin,
},
rules: [
  ...jestPlugin.configs['flat/recommended'].rules,
  ...testingLibraryPlugin.configs['flat/react'].rules,
]
```

>or are you suggesting we'd have to do something liek this

Almost ;) 
In your code, the last `rules` block would overwrite `...jestPlugin['configs']`, `...testingLibraryPlugin.configs` etc, since it's a regular JS object. 

I enabled all recommended configs for our plugins. Previously we didn't do it for the react plugin, so I had to disable a few rules (I made them warnings).